### ISSUE TITLE
PHP Fatal error when calling stream destructors

### ISSUE THREAD

**Issue by klammbueddel** [guzzle/guzzle#3005]
**Guzzle version(s) affected**: 7.4.2
**PHP version**: 8.1.4
**cURL version**: 7.74.0

**Description**  
On our production server we see a lot of the following errors:

`PHP Fatal error: Couldn't execute method GuzzleHttp\\Handler\\CurlMultiHandler::__destruct in Unknown on line 0`
`Couldn't execute method GuzzleHttp\\Psr7\\Stream::__destruct in Unknown on line 0`

We still have no clue under which circumstances this happens. We do not see this error in our dev / staging environment, which is kind of odd and makes debugging really hard. We use AWS SDK in general, but the requests which produces the error do not access AWS in this context. Maybe somebody has a clue what's going on here and what we could try to analyze it. The request itself will pass our application controller, the final log which is written in the shutdown_function says everything was fine. The error seems to occur when PHP calls all the destructors, the load balancer then responds with 502 Bad Gateway error.

**How to reproduce**  
Still hard to tell..

**Comment by SachinBahukhandi** [comment]
@klammbueddel were you able to replicate?

**Comment by klammbueddel** [comment]
No, we had to rollback to PHP 7.4 in order to fix this.

**Comment by mfn** [comment]
> in Unknown on line 0

This is weird. Do you by chance run with opcache enabled? If so, can you try disabling it on PHP8?

**Comment by klammbueddel** [comment]
No, opcache was not yet enabled.

**Comment by xnan-dev** [comment]
Any idea how this could be somehow reproduced ?

**Comment by klammbueddel** [comment]
No, I hoped anybody could help or maybe saw this in other context. I found nothing related in the internet, so I opened this issue.

**Comment by mmaglaya** [comment]
Hi, I face the same issue when I upgraded to PHP 8 and trying to fix it since few days. Just found this open issue. 

PHP 8.0.20
Guzzle 7.5.0
Curl 7.74.0

**Comment by haackker68** [comment]
I am facing the same issue as well since I upgraded to PHP 8.1. I have tried multiple solutions but nothing seemed to work. 

Tried upgrading to PHP 8.2 but no luck. 


> > in Unknown on line 0
> 
> This is weird. Do you by chance run with opcache enabled? If so, can you try disabling it on PHP8?

I had tried disabling this but still no change.

**Comment by usmyum** [comment]
Same here! The issue for me came exactly at the same point when I upgraded to Php 8.1 and no solutions are working for me so far. 

Looking for support.

**Comment by umair403** [comment]
Hi guys! I have upgraded to php-8 and start facing this issue. I think issue is with Guzzlehttp GuzzleHttp\\Handler\\CurlMultiHandler  we had to roll back to previous version of php.  anyone is able to fix this please reply.

**Comment by klammbueddel** [comment]
We still have this issue, but I want to share some findings with you:

In our code, we registered the Aws stream wrapper with Aws\S3\StreamWrapper::register or Aws\S3\S3Client::registerStreamWrapper in every request. Once we removed this and only register the stream wrapper on demand, we significantly reduced the error count and finally could switch to PHP 8.2.

Still, we see the error even in requests, where no stream wrappers are registered. We are running our code in a Kubernetes cluster. We see significant more errors during deployments, when Pods are terminated and recreated and in the minutes / hours after a deployment the error count fades out.

Maybe anybody has a clue what is going on here. Could it be some corrupt memory, where  PHP is trying to destroy objects that never exists in this context but maybe did in another?
